actions-server/src/handlers/joinGame.ts

```diff
@@ -1,90 +1,74 @@
 
 // Request Handler
 const handler = async (req: Request, res: Response) => {
+  if (!process.env.HASURA_ENDPOINT || !process.env.HASURA_GRAPHQL_JWT_SECRET) {
+    return res.status(500)
+  }
+  const client = new GraphQLClient(process.env.HASURA_ENDPOINT)
+  const sdk = getSdk(client)
+
   // get request input
   const { gameId, clientUuid } = req.body.input
 
   // execute the Hasura operation(s)
   let playerId
-  const { data: lookupData, errors } = await execute<
-    LookupPlayerForGameQueryVariables
-  >(LookupPlayerForGame, {
-    gameId,
-    clientUuid,
-  })
-  if (errors) {
-    return res.status(400).json(errors[0])
-  }
-  if (lookupData.players[0]) {
-    // already joined game
-    playerId = lookupData.players[0].id
-  } else {
-    // new player for game
-    const { data: insertData, errors } = await execute<
-      InsertPlayerForGameMutationVariables
-    >(InsertPlayerForGame, {
+
+  try {
+    const { players } = await sdk.LookupPlayerForGame({
       gameId,
       clientUuid,
     })
-    if (errors) {
-      return res.status(400).json(errors[0])
+    if (players[0]) {
+      // already joined game
+      playerId = players[0].id
+    } else {
+      // new player for game
+      try {
+        const { insert_players_one } = await sdk.InsertPlayerForGame({
+          gameId,
+          clientUuid,
+        })
+        if (insert_players_one) {
+          playerId = insert_players_one.id
+        }
+      } catch {
+        return res.status(400)
+      }
     }
-    playerId = insertData.insert_players_one.id
-  }
 
-  if (!playerId) {
-    return res.status(500)
-  }
+    if (!playerId) {
+      return res.status(500)
+    }
 
-  const tokenContents = {
-    sub: playerId.toString(),
-    iat: Date.now() / 1000,
-    iss: "https://fishbowl-game.com/",
-    "https://hasura.io/jwt/claims": {
-      "x-hasura-allowed-roles": ["player", "anonymous"],
-      "x-hasura-user-id": playerId.toString(),
-      "x-hasura-default-role": "player",
-      "x-hasura-role": "player",
-    },
-    exp: Math.floor(Date.now() / 1000) + 24 * 60 * 60,
-  }
+    const tokenContents = {
+      sub: playerId.toString(),
+      iat: Date.now() / 1000,
+      iss: "https://fishbowl-game.com/",
+      "https://hasura.io/jwt/claims": {
+        "x-hasura-allowed-roles": ["player", "anonymous"],
+        "x-hasura-user-id": playerId.toString(),
+        "x-hasura-default-role": "player",
+        "x-hasura-role": "player",
+      },
+      exp: Math.floor(Date.now() / 1000) + 24 * 60 * 60,
+    }
 
-  const token = jwt.sign(
-    tokenContents,
-    process.env.HASURA_GRAPHQL_JWT_SECRET || "missing secret"
-  )
+    const token = jwt.sign(
+      tokenContents,
+      process.env.HASURA_GRAPHQL_JWT_SECRET || "missing secret"
+    )
 
-  return res.json({
-    id: playerId.toString(),
-    jwt_token: token,
-  })
+    return res.json({
+      id: playerId.toString(),
+      jwt_token: token,
+    })
+  } catch {
+    return res.status(400)
+  }
```

Comment:
looking for a better way to handle the errors, i should have access to the "error" key when this happens.